### ISSUE TITLE
response_to_ldif() failing

### ISSUE THREAD

**Issue by slushpupie** [cannatag/ldap3#896]
I cant seem to get the `response_to_ldif()` call to work correctly. 

In a trivial case search case that is successful and has a single result, I get this error:

```
>>> c.response_to_ldif()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/venv/lib/python3.9/site-packages/ldap3/core/connection.py", line 1441, in response_to_ldif
    ldif_lines = operation_to_ldif('searchResponse', search_result, all_base64, sort_order=sort_order)
  File "/venv/lib/python3.9/site-packages/ldap3/protocol/rfc2849.py", line 221, in operation_to_ldif
    lines = search_response_to_ldif(entries, all_base64, sort_order)
  File "/venv/lib/python3.9/site-packages/ldap3/protocol/rfc2849.py", line 135, in search_response_to_ldif
    raise LDAPLDIFError('unable to convert to LDIF-CONTENT - missing DN')
ldap3.core.exceptions.LDAPLDIFError: unable to convert to LDIF-CONTENT - missing DN
```

Looking at that function, it seems this line is suspect: 

https://github.com/cannatag/ldap3/blob/v2.8.1/ldap3/protocol/rfc2849.py#L132
```python
lines.append(_convert_to_ldif('dn', entry['dn'], all_base64))
```

Since the DN isnt an attribute, that dosnt work right.  Shouldn't it look something like:

```python
lines.append(_convert_to_ldif('dn', entry.distinguishedName, all_base64))
```

(I'm guessing here)

**Comment by slushpupie** [comment]
Hmm.  so it looks like the entry itself _does_ have a dn key.  However, the list of entries contains more than just the searchResEntry, it also has a searchResRef, which _does not_ have a DN and thus fails here.

**Comment by cannatag** [comment]
HI, i could limit the ldif to the searchResEntry, because the searchResRef has no representation in the LDIF format.

**Comment by cannatag** [comment]
Can you check the code in dev? It now converts only searchResEntry entries to LDIF.

**Comment by slushpupie** [comment]
So now `c.response_to_ldif()` returns a `None` instead of an error.  If I call `c.response_to_ldif(response)` explicitly I get the same exception.  

Wouldn't this function need to be changed to ignore non-`searchResEntry` objects?

https://github.com/cannatag/ldap3/blob/dev/ldap3/protocol/rfc2849.py#L125-L143

**Comment by cannatag** [comment]
Fixed in ldap3 v2.9
